internal/connection/connection_pool.go

```diff
@@ -82,34 +84,49 @@ func (cp *ConnectionPool) Delete(key string) {
 
 // getByDriverName returns map of connections filtered by driverName. This function
 // must be called with read lock held.
-func (cp *ConnectionPool) getByDriverName(driverName string) map[string]*Connection {
+func (cp *ConnectionPool) getByDriverName(driverName string) (map[string]*Connection, error) {
 	newPool := make(map[string]*Connection)
 	for k, v := range cp.pool {
 		if v.DriverName != driverName {
 			continue
 		}
+		if !v.connected {
```

Comment:
This if-block is repeated quite a few times. It would be nicer to put it in an `Connection.EnsureConnected()` function, or something like that.

Reply:
Done!